scripts/create.sh

```diff
@@ -25,78 +49,117 @@ check_task() {
   t=$(get_task $taskId)
 
   state=$(echo $t | jq -r .status)
+  echo "... current status [${state}]"
+  echo "... error: $(echo $t | jq .response.error.description)"
+
+  if [ "$state" = "processing-error" ]; then
+      exit 0
+  fi
 
-  if [[ "$state" != "processing-completed" ]]; then
+  if [ "$state" != "processing-completed" ]; then
       return 1
   fi
 
   return 0
 }
 
-wait_for_db() {
-  echo "to be used to wait for the database public endpoint"
-}
-
 check_redis() {
-  echo "-> checking redis using ping"
-  redis-cli -u redis://default:${DB_PASS}@${DB_HOST}:${DB_PORT} ping 2>/dev/null
+  host="$1"
+  port="$2"
+  res=$(redis-cli -u redis://default:${DB_PASS}@${host}:${port} ping 2>/dev/null)
+  echo $res
+  if [ "$res" = "PONG" ]; then
+    return 0
+  else
+    return 1
+  fi
 }
 
-# Get list of available plans
-echo "-> about to get list of available plans"
-plans=$(curl -sX GET \
-  -H "x-api-key: ${ACCOUNT_KEY}" \
-  -H "x-api-secret-key: ${SECRET_KEY}" \
-  ${API_URL}/${PLAN_TYPE}/plans)
-
-# Get planId from provider, region and name
-planId=$(echo $plans | jq \
-  --arg region "$REGION" \
-  --arg provider "$PROVIDER" \
-  --arg plan_name "$PLAN_NAME" \
-  '.plans[] | select(.name == $plan_name and .provider == $provider and .region == $region) | .id')
-echo "-> selected planId is [$planId]"
-
-# Make sure a plan was retrieved
-if [[ "$planId" = "" ]]; then
-  echo "-> planId is empty. Aborting..."
-  exit 1
-fi
+########################
+###  Creation logic  ###
+########################
 
-# Create a subscription
-echo "-> about to create a Redis Cloud subscription"
-task=$(curl -sX POST \
-  -H "Content-Type: application/json" \
-  -H "x-api-key: ${ACCOUNT_KEY}" \
-  -H "x-api-secret-key: ${SECRET_KEY}" \
-  -d "{
-    \"name\": \"My new subscription\",
-    \"planId\": $planId
-  }" \
-  ${API_URL}/${PLAN_TYPE}/subscriptions)
-taskId=$(echo $task | jq -r .taskId)
-echo "-> associated task is [$taskId]"
+# If no subscription is provided, one will be created using those values
+# (will be used as service configuration in next versions)
+PLAN_TYPE="fixed"
+PLAN_NAME="Standard 30MB"
 
-# Waiting for subscription task to be completed"
-while ! check_task "$taskId"; do
-  echo "-> waiting for subscription task to be completed"
-  sleep 2
-done
-echo "-> Subscription task completed"
+# Internal subscription related variables
+planType=${PLAN_TYPE}
+subscriptionId=${SUBSCRIPTION_ID}
+
+# Keep track of the subscription id is created alongside the database
+internalSubscription=""
+
+# Retrieve subscription if id provided or create a new one
+if [[ "${subscriptionId}" != "" ]]; then
+  # Get the subscription's type
+  type=$(get_subscription_type ${subscriptionId})
+  if [ "$type" != "invalid" ]; then
+    planType=$type
+  else
+    echo "-> provided subscription id [${subscriptionId}] is invalid"
+    exit 0
```

Comment:
Exit 1, this runs and the job never runs again. If the user made a mistake, they can't fix it.

Reply:
Ok I'll also make sure an exit 1 is used when a task (related to subscription or database creation) raises an error.